cpp/include/cuml/cluster/linkage.hpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2018-2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/linalg/distance_type.h>
+#include <raft/sparse/hierarchy/common.h>
+
+#include <cuml/cuml.hpp>
+
+namespace ML {
+
+/**
+ * @defgroup HdbscanCpp C++ implementation of Dbscan algo
```

Comment:
Just realized I should probably add the correct docs here and doc both pairwise and knn functions